lib/openstudio-standards/standards/ashrae_90_1/ashrae_90_1_2007/179d_ashrae_90_1_2007/179d_ashrae_90_1_2007.Model.rb

```diff
@@ -709,6 +749,209 @@ def model_create_prm_any_baseline_building(user_model, building_type, climate_zo
+            
+            # Check if adjustment is needed (more than 5% difference)
+            if total_proposed_oa > 0 && (current_total_air_loop_oa - total_proposed_oa).abs > total_proposed_oa * 0.05
+              diff_pct = ((current_total_air_loop_oa - total_proposed_oa).abs / total_proposed_oa) * 100
+              puts "*** DEBUG PUTS: Air loop OA difference is #{diff_pct.round(2)}% ***"
+              OpenStudio.logFree(OpenStudio::Info, 'openstudio.standards.Model', "Air loop OA difference is #{diff_pct.round(2)}%, adjusting air loop OA flow rates")
+              
+              # Handle two cases: proportional adjustment vs. distribution from zero
+              if current_total_air_loop_oa > 0
+                # Case 1: Proportional adjustment when baseline already has OA flow rates
+                adjustment_factor = total_proposed_oa / current_total_air_loop_oa
+                puts "*** DEBUG PUTS: Using proportional adjustment factor: #{adjustment_factor.round(4)} ***"
+                
+                # Apply proportional adjustment to each air loop
+                air_loops.each do |air_loop|
+                  next if air_loop.airLoopHVACOutdoorAirSystem.empty?
+                  
+                  oa_system = air_loop.airLoopHVACOutdoorAirSystem.get
+                  controller_oa = oa_system.getControllerOutdoorAir
+                  sizing_system = air_loop.sizingSystem
+                  
+                  # Get current OA flow rate for this air loop
+                  current_air_loop_oa = 0.0
+                  if controller_oa.minimumOutdoorAirFlowRate.is_initialized
+                    current_air_loop_oa = controller_oa.minimumOutdoorAirFlowRate.get
+                  elsif sizing_system.designOutdoorAirFlowRate.is_initialized
+                    current_air_loop_oa = sizing_system.designOutdoorAirFlowRate.get
+                  end
+                  
+                  # Calculate new OA flow rate for this air loop
+                  new_air_loop_oa = current_air_loop_oa * adjustment_factor
+                  
+                  # Set the new values - ensure maximum is at least equal to minimum to avoid EnergyPlus constraint violations
+                  sizing_system.setDesignOutdoorAirFlowRate(new_air_loop_oa)
+                  controller_oa.setMinimumOutdoorAirFlowRate(new_air_loop_oa)
+                  # Set maximum to be at least equal to minimum to satisfy EnergyPlus constraints
+                  if controller_oa.maximumOutdoorAirFlowRate.is_initialized
+                    current_max = controller_oa.maximumOutdoorAirFlowRate.get
+                    if current_max < new_air_loop_oa
+                      controller_oa.setMaximumOutdoorAirFlowRate(new_air_loop_oa)
+                    end
+                  else
+                    controller_oa.setMaximumOutdoorAirFlowRate(new_air_loop_oa)
```

Comment:
and do we ever want this.. setting min required OA to max OA rate in OA controller. I'm actually unsure.